tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java

```diff
@@ -25,23 +35,54 @@
 public class BeanDiscoveryInArchivesTemplate extends CDITestBase {
 
 	@Before
-	public void setupProject() {
-		createClass(PROJECT_NAME, "test", "CdiBean1");
-		createClass(PROJECT_NAME, "test", "CdiBean2");
-		ProjectExplorer pe = new ProjectExplorer();
-		pe.open();
-		pe.getProject(PROJECT_NAME).getProjectItem(CDIConstants.JAVA_RESOURCES, CDIConstants.SRC, "test", "CdiBean2.java").open();
-		TextEditor ed = new TextEditor("CdiBean2.java");
-		ed.insertLine(1, "import javax.inject.Inject;");
-		ed.insertLine(2, "import javax.enterprise.context.RequestScoped;");
-		ed.insertLine(4, "@RequestScoped");
-		ed.insertLine(6, "@Inject CdiBean1 bean1;");
-		ed.save();
+	public void setupProjectAccordingToGivenValues() {
+		valuesToInsertDuringProjectSetup.put(1, "import javax.inject.Inject;");
+		valuesToInsertDuringProjectSetup.put(2, "import javax.enterprise.context.RequestScoped;");
+		valuesToInsertDuringProjectSetup.put(4, "@RequestScoped");
+		valuesToInsertDuringProjectSetup.put(6, "@Inject CdiBean1 bean1;");
+		super.setupProject(valuesToInsertDuringProjectSetup);
 	}
 	
 	@After
 	public void clean() {
 		deleteAllProjects();
 	}
 	
+	public void validationOfBeanDiscoveryInExplicitArchives() {
+		prepareBeanXml("all", false);
+		ProjectExplorer pe = new ProjectExplorer();
+		pe.open();
+		pe.getProject(PROJECT_NAME).getProjectItem(CDIConstants.JAVA_RESOURCES, CDIConstants.SRC, "test", CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION).open();
+		TextEditor ed = new TextEditor(CDI_BEAN_2_JAVA_FILE_NAME + JAVA_FILE_EXTENSION);
+		ed.insertLine(7, "@Inject String warningHere;");
+		ed.save();
+		try{
+			new WaitUntil(new EditorHasValidationMarkers(ed));
+		} catch (WaitTimeoutExpiredException ex){
+			fail("There is supposed to be warning present");
+		}
+		List<Marker> markers = ed.getMarkers();
+		assertEquals(1, markers.size());
+		Marker validation = markers.get(0);
+		assertEquals(validation.getLineNumber(), 8);
+		assertTrue(validation.getText().contains("No bean is eligible for injection"));
```

Comment:
We will need to improve the message/problem check. Here was checked only problem's short message, now we have to distinguish between CDI11 and CDI20 - JSR-346 and JSR-365 specs. Some validator would solve the issue well.

Reply:
Fixed.